data/data/bootstrap/files/usr/local/bin/bootkube.sh.template

```diff
@@ -230,6 +231,45 @@ then
     record_service_stage_success
 fi
 
+if [ ! -f cloud-controller-manager-bootstrap.done ]
+then
+    record_service_stage_start "cluster-cloud-controller-manager-bootstrap"
+	echo "Rendering Cloud Controller Manager core manifests..."
+
+	rm --recursive --force cloud-controller-manager-bootstrap
+
+	# Copy the CCCMO images configMap to resolve pod images from internal registry in CCCMO render run
+	# Source config map is located in https://github.com/openshift/cluster-cloud-controller-manager-operator/blob/master/manifests/0000_26_cloud-controller-manager-operator_01_images.configmap.yaml
+	bootkube_podman_run \
+	--name copy-cloud-controller-manager-images \
+	--entrypoint /bin/bash "${RELEASE_IMAGE_DIGEST}" \
+	-c "cat /release-manifests/0000_26_cloud-controller-manager-operator_01_images.configmap.yaml" > manifests/ccm-images.yaml
+
+	ADDITIONAL_FLAGS=""
+	if [ -f "$PWD/manifests/cloud-provider-config.yaml" ]; then
+		ADDITIONAL_FLAGS="--cloud-config-file=/assets/manifests/cloud-provider-config.yaml"
+	fi
+
+	bootkube_podman_run \
+		--name cloud-controller-render \
+		--volume "$PWD:/assets:z" \
+		"${CLUSTER_CLOUD_CONTROLLER_MANAGER_OPERATOR_IMAGE}" \
+		/render run \
+		--images-file=/assets/manifests/ccm-images.yaml \
+		--dest-dir=/assets/cloud-controller-manager-bootstrap \
+		--cluster-infrastructure-file=/assets/manifests/cluster-infrastructure-02-config.yml \
+		${ADDITIONAL_FLAGS}
+
+	# Copy rendered resources to manifests folder
+	cp cloud-controller-manager-bootstrap/bootstrap/* bootstrap-manifests/
+	# Copy cloud config to /etc/kubernetes/bootstrap-configs
+	cp cloud-controller-manager-bootstrap/config/* /etc/kubernetes/bootstrap-configs
```

Comment:
this will fail if no cloud config is generated, when i run on aws and there is no cloud config, i see this in the bootkube.service output:

```
Jul 01 13:29:18 ip-10-0-12-89 bootkube.sh[2207]: cp: cannot stat 'cloud-controller-manager-bootstrap/config/*': No such file or directory
```

Reply:
this patch is allowing me to proceed
```
diff --git a/data/data/bootstrap/files/usr/local/bin/bootkube.sh.template b/data/data/bootstrap/files/usr/local/bin/bootkube.sh.template
index a36dbee28..3c3c61d6c 100755
--- a/data/data/bootstrap/files/usr/local/bin/bootkube.sh.template
+++ b/data/data/bootstrap/files/usr/local/bin/bootkube.sh.template
@@ -245,8 +245,10 @@ then
        -c "cat /release-manifests/0000_26_cloud-controller-manager-operator_01_images.configmap.yaml" > manifests/ccm-images.yaml
 
        ADDITIONAL_FLAGS=""
+    CLOUD_CONFIG_EXISTS=""
        if [ -f "$PWD/manifests/cloud-provider-config.yaml" ]; then
                ADDITIONAL_FLAGS="--cloud-config-file=/assets/manifests/cloud-provider-config.yaml"
+        CLOUD_CONFIG_EXISTS="yes"
        fi
 
        bootkube_podman_run \
@@ -262,7 +264,9 @@ then
        # Copy rendered resources to manifests folder
        cp cloud-controller-manager-bootstrap/bootstrap/* bootstrap-manifests/
        # Copy cloud config to /etc/kubernetes/bootstrap-configs
-       cp cloud-controller-manager-bootstrap/config/* /etc/kubernetes/bootstrap-configs
+    if [ ! -z "$CLOUD_CONFIG_EXISTS" ]; then
+        cp cloud-controller-manager-bootstrap/config/* /etc/kubernetes/bootstrap-configs
+    fi
 
 
        touch cloud-controller-manager-bootstrap.done
```

Reply:
Ahh... Bash weirdness. Apparently this throws an error you saw if the directory is empty, even if there is wildcard at the source for the copy.
```bash
cp cloud-controller-manager-bootstrap/config/* /etc/kubernetes/bootstrap-configs
```

This works instead, even if the directory is empty:
```bash
cp -r cloud-controller-manager-bootstrap/config/. /etc/kubernetes/bootstrap-configs
```